Location propagation in case of non-existing resource group

mikhailshilkov opened the pull request:
Fixes #239

mikhailshilkov commented:
So, the issue is that we call `.StringValue` on a computed property, which fails at preview time.

I added dumb checks and return a hard-coded value `[computed]` in this case, which prevents the runtime error. The resources are created successfully and the location is populated.

Obviously, `[computed]` shows up in the preview, which is ugly:

![afbeelding](https://user-images.githubusercontent.com/1454008/57632767-923af580-75a2-11e9-8cdd-3e1305f697af.png)

Not sure if it also breaks anything.

My knowledge at the point is not enough to pull the value out of the output. Hints are very welcome!

lukehoban commented on resources.go:
```diff
@@ -964,27 +964,32 @@ func Provider() tfbridge.ProviderInfo {
-										} else {
-											rgRegion = azure.NormalizeLocation(rgData.Get("location"))
+									if rg.IsComputed() || rg.IsOutput() {
+										return "[computed]", nil
```
I *think* that it would be better to not magic up a value here - and to instead just ensure that we return `config.UnknownVariableValue`.  That should get replaced with something like `[computed]` at the CLI rendering layer - but will ensure that the value flows correctly through other layers.

mikhailshilkov replied:
Great suggestion! This prints `output<string>` in the preview details.

mikhailshilkov commented:
@lukehoban Great idea! There was one example without `location` but it wasn't a part of the test suite. I've added it to the fast list, and removed `location` from a number of other tests. However, it seems that CI is not configured properly, as I see this in the build log:

```
--- SKIP: TestExamples//home/travis/gopath/src/github.com/pulumi/pulumi-azure/examples/minimal (0.00s)
        program.go:465: Skipping: PULUMI_ACCESS_TOKEN is not set
...
```

lukehoban commented:
>  However, it seems that CI is not configured properly, as I see this in the build log

Yes - TravisCI builds do not pick up credentials when run in forks unfortunately (security reasons).  So to run these tests you would need to re-open the PR as a branch in the main repo.  Want to do that to get a test run?

lukehoban commented:
Merged #248 with the same changes, so closing this one out.
